packages/@sanity/vision/src/components/VisionGui.tsx

```diff
@@ -1027,3 +1110,62 @@ export class VisionGui extends PureComponent<VisionGuiProps, VisionGuiState> {
     )
   }
 }
+
+function getActivePerspective({
+  visionPerspective,
+  pinnedPerspective,
+}: {
+  visionPerspective: ClientPerspective | SupportedPerspective
+  pinnedPerspective: PerspectiveValue
+}): ClientPerspective | undefined {
+  if (visionPerspective !== 'pinnedRelease') {
+    return visionPerspective
+  }
+
+  if (pinnedPerspective.perspectiveStack.length !== 0) {
+    return pinnedPerspective.perspectiveStack
```

Comment:
My question on the previous comment ☝️ is due to this, it looks like we might actually just need 
perspectiveStack
selectedPerspectiveName
selectedPerspective,

or maybe I'm missing something? 😬